packages/discovery-react-components/src/components/StructuredQuery/StructuredQuery.tsx

```diff
@@ -19,41 +20,49 @@ export interface StructuredQueryProps {
 
 const StructuredQuery: FC<StructuredQueryProps> = ({ messages = defaultMessages }) => {
   const mergedMessages = { ...defaultMessages, ...messages };
-  const [ruleRows, setRuleRows] = useState({ rows: [{ id: 0 }] });
-  const showAddRuleRowButton = ruleRows.rows.length < MAX_NUM_SIBLING_RULE_ROWS;
+  const [structuredQuerySelection, setStructuredQuerySelection] = useState<
+    StructuredQuerySelection
+  >({
+    groups: {
+      0: { rows: [0] }
+    }
+  });
 
-  const handleAddRuleRowOnClick = () => {
-    const newRuleRowId = ruleRows.rows[ruleRows.rows.length - 1].id + 1;
-    const newRuleRow = { id: newRuleRowId };
-    setRuleRows({
-      ...ruleRows,
-      rows: ruleRows.rows.concat(newRuleRow)
-    });
-  };
+  const showAddRuleRowButton =
+    structuredQuerySelection.groups[0].rows.length < MAX_NUM_SIBLING_RULE_ROWS;
+  const showAddRuleGroupButton =
+    keys(structuredQuerySelection.groups).length - 1 < MAX_NUM_NESTED_RULE_GROUPS;
```

Comment:
is there a difference between `keys` and `Object.keys`?

Reply:
I thought there was a reason I used that instead, but actually I think there isn't a difference, so updated to just use `Object.keys` 👍